azerothcore/azerothcore-wotlk
# fix(Core/Players): Fixed getting rested exp in cities/inns.

Fixes #7939



## Issues Addressed:

- Closes #7939
- Closes https://github.com/chromiecraft/chromiecraft/issues/1773

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

1. Find a char that is known to gain rested xp when offline.
2. Use a command to check rested xp. (I just copy/paste the macro for reporting rested xp. `/script p="player";x=UnitXP(p);m=UnitXPMax(p);r=GetXPExhaustion();if -1==(r or -1)then t="No rest."else t="Rest: "..(math.floor(20*r/m+0.5)).." bubbles ("if r+x<m then t=t..r else t=t.."level +"..(r+x-m)end t=t.."XP)"end;DEFAULT_CHAT_FRAME:AddMessage(t)).`
3. Go to a main city/inn, check portrait for glow and "zzz". (Both should look OK.)
4. Wait for a few minutes. If the "xp to next level" for the char's current level is large enough (like for a char above lvl 15), even a few minutes should provide observable (>=1) rested xp (assuming one bubble per 8 hr).
5. Use a command to check rested xp again. Compare with the value before entering the city/inn.
6. You should get blue indicator and bar informing about rested exp.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
